core/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -199,9 +192,6 @@ protected static boolean parseObjectOrDocumentTypeProperties(String fieldName, O
                     parseProperties(builder, (Map<String, Object>) fieldNode, parserContext);
                 }
                 return true;
-            } else if (fieldName.equals("include_in_all")) {
-                builder.includeInAll(TypeParsers.nodeBooleanValue(fieldName, "include_in_all", fieldNode, parserContext));
```

Comment:
hmm I don't like the fact that we did not fail when it was set with 6.x indices. Maybe we should issue a deprecation warning and ignore the `include_in_all` value?